signer/utils.go

```diff
@@ -0,0 +1,114 @@
+package signer
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+)
+
+// GetKeySpec picks up a recommended signing algorithm for given certificate.
+func GetKeySpec(signingCert *x509.Certificate) (KeySpec, error) {
+	var keyspec KeySpec
+	switch key := signingCert.PublicKey.(type) {
+	case *rsa.PublicKey:
+		switch key.Size() {
+		case 256:
+			keyspec = RSA_2048
+		case 384:
+			keyspec = RSA_3072
+		case 512:
+			keyspec = RSA_4096
+		default:
+			return "", UnsupportedSigningKeyError{keyType: "rsa", keyLength: key.Size()}
+		}
+	case *ecdsa.PublicKey:
+		switch key.Curve.Params().BitSize {
+		case 256:
+			keyspec = EC_256
+		case 384:
+			keyspec = EC_384
+		case 521:
+			keyspec = EC_521
+		default:
+			return "", UnsupportedSigningKeyError{keyType: "ecdsa", keyLength: key.Curve.Params().BitSize}
+		}
+	}
+	return keyspec, nil
+}
+
+// GetLocalSignatureProvider returns the LocalSignatureProvider created using given certificates and private key.
+func GetLocalSignatureProvider(certs []*x509.Certificate, pk crypto.PrivateKey) (*LocalSignatureProvider, error) {
```

Comment:
```suggestion
func NewLocalSignatureProvider(certs []*x509.Certificate, pk crypto.PrivateKey) (*LocalSignatureProvider, error) {
```

Reply:
As this will require updating notaiton-go-lib will do in next PR with signature scheme changes.